gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -693,14 +736,21 @@ private void requestObjectMedia(OptionalLong bytesToRead) throws StatusRuntimeEx
     StorageBlockingStub blockingStub = stub.withDeadlineAfter(readTimeout, MILLISECONDS);
     try {
       if (useZeroCopyMarshaller) {
-        resIterator =
-            io.grpc.stub.ClientCalls.blockingServerStreamingCall(
+        Iterator<ReadObjectResponse> responseIterator =
```

Comment:
Shouldn't we add this logic to readFooter as well? normal read and footer read implementation is diverging.

Reply:
Not adding to footer was intentional.I agree,We should merge the impl for footer and read api. (Zero-copy deserializer and Watch dog impl )